pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -361,3 +373,8 @@ def __log_result_step_metrics(self, step_log_metrics, step_pbar_metrics, batch_i
 
         if len(step_pbar_metrics) > 0:
             self.trainer.logger_connector.add_progress_bar_metrics(step_pbar_metrics)
+
+    def _set_batch_indices(self, dataloader_idx: int, batch_samplers):
+        batch_sampler = batch_samplers[dataloader_idx]
+        batch_indices = batch_sampler.batch_indices if isinstance(batch_sampler, LightningBatchSamplerWrapper) else None
+        self.trainer.batch_indices = batch_indices
```

Comment:
must this really be a trainer property? Can't this live in the evaluation loop and the trainer fetches it from there. We are polluting the trainer with more and more attributes...

I feel an internal variable in the eval_loop that is accessed by the trainer should be sufficient and it must not be exposed at all.

Reply:
Definitely right ! Done !